lib/mongoid/extensions/big_decimal.rb

```diff
@@ -58,7 +68,17 @@ def demongoize(object)
         #
         # @return [ String, nil ] A String representing the object or nil.
         def mongoize(object)
-          object && object.numeric? ? object.to_s : nil
+          if object
+            if Mongoid.map_big_decimal_to_decimal128
+              if object.is_a?(BigDecimal)
+                BSON::Decimal128.new(object)
+              elsif object.numeric?
```

Comment:
If this method (attempts) to handle a variety of input types, it seems to me that the second condition of this if statement (ahead of `object.numeric?` branch) should be `elsif object.is_a?(BSON::Decimal128)` which would return `object` verbatim.

Reply:
done